dora/core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -314,9 +314,9 @@ public void cache(CacheRequest request) {
   }
 
   @Override
-  public ListenableFuture<CacheDataResponse> cacheData(CacheDataRequest request) {
+  public CacheDataResponse cacheData(CacheDataRequest request) {
     try {
-      return mRpcFutureStub.withDeadlineAfter(mRpcTimeoutMs, TimeUnit.MILLISECONDS)
+      return mRpcBlockingStub.withDeadlineAfter(mRpcTimeoutMs, TimeUnit.MILLISECONDS)
```

Comment:
why changed to a blocking implementation?

Reply:
we want to make sure the first call finishes before sending the second call other the tasks in the worker queue will be disordered.

Reply:
in that case you can instead wait on the returned future instead of making it blocking.